### PR TITLE
Check if module is already loaded in LoadSymbols

### PR DESCRIPTION
**Pull request by beckerhe**
This avoids a race condition that occurs when a user requests to load
the symbols of a module again while the previous request is still
being processed.

The additional check avoids the race condition and hence the crash.

Bug: http://b/205002963

### CODE REVIEW

File: src/OrbitGl/App.cpp [hunk 1]
```diff
@@ -1889,6 +1889,11 @@ orbit_base::Future<ErrorMessageOr<void>> OrbitApp::LoadSymbols(
     return it->second;
   }
 
+  const ModuleData* module_data = GetModuleByPathAndBuildId(module_file_path, module_build_id);
+  if (module_data->is_loaded()) {
+    return ErrorMessageOr<void>{outcome::success()};
```
Comment (dpallotti) [inline src/OrbitGl/App.cpp:1894]: I would add an `ERROR` though, and a comment pointing to the bug. At first glance this shouldn't be possible.

**Review by dpallotti — COMMENTED** [review]
I don't know the code a lot. I tried to look into it a bit and read the bug carefully. Is http://b/205002963#comment4 not feasible or too complex? While this fixes the crash, potentially calling `RetrieveModule` again seems like the same kind of wrong, with the only difference that the `CHECK` in `AddSymbols` isn't hit because it comes later.

**Comment by beckerhe** [comment]
> I don't know the code a lot. I tried to look into it a bit and read the bug carefully. Is http://b/205002963#comment4 not feasible or too complex? While this fixes the crash, potentially calling `RetrieveModule` again seems like the same kind of wrong, with the only difference that the `CHECK` in `AddSymbols` isn't hit because it comes later.

It's feasible but more work than expected, assuming we don't want to hack it in which would result in even more Spaghetti-code. And since we wanna refactor this part anyway soon, I opted for making a minimal change that only avoids the crash.

And yes, this solution does not fix the problem that RetrieveModule is called twice. But we have to consider that it's also very unlikely to hit that race condition. You really need to trigger the second load request in the exact right moment. I wasn't able to reproduce it without adding sleeps in multiple locations. Pablo tried as well and wasn't able either.

File: src/OrbitGl/App.cpp (hunk 1 above)
Comment (beckerhe) [reply src/OrbitGl/App.cpp:1894]: Yep. Will do.

Comment (beckerhe) [reply src/OrbitGl/App.cpp:1898]: Done.

**Review by dpallotti — APPROVED** [review]

**Pull request merged**